src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1461,7 +1473,7 @@ private void Abort(Exception abortException)
 
             // Check idle timeout when there are not pending requests for a while.
             if ((connectionIdleTimeout != Timeout.InfiniteTimeSpan) &&
-                (_httpStreams.Count == 0) &&
+                (_httpStreams.Count == 0) && (_idleSinceTickCount > 0) &&
```

Comment:
Why is this change necessary?

Reply:
On a new Http2Connection `_idleSinceTickCount` is set to 0 until the first request is sent. On the other hand, setting `PooledConnectionIdleTimeout` to a specific value starts a timer calling this `CleanCacheAndDisposeIfUnused` method every second or less, so this check gets done before the first request goes through connection while `_idleSinceTickCount` is still 0 which is always less than`Environment.TickCount64` therefore a brand new connection can get almost immediately expired.

Reply:
So this issue existed previously? I'm glad you caught it!

That said, maybe it would be better to initialize `_idleSinceTickCount` to the current tick in the Http2Connection constructor? I'm not sure where we check this value, but it seems better to have it be correct always than to special case 0.

Adding @wfurt, I think he wrote this code... thoughts?

Reply:
Yes, it is an existing issue. Currently `_idleSinceTickCount` is read only in this method and assigned in `RemoveStream`. We could initialize it in the constructor, but it seems the intention was to update it when a request completes and stream gets closed, thus setting a value in the ctor will break that invariant.

Reply:
The intent is to track the last time the connection might have been idle (if there were no other active streams). As such initializing it to the current tick count in the constructor seems to be more inline with the intent, to me.

I don't know if there's ever a case where we can get into a situation where an Http2Connection is created but never actually used -- maybe if cancellation happens at exactly the right time? If that ever happens, we will still want to idle the connection out eventually.

Reply:
Agree, makes sense. I will initialize it in the constructor.